### DISCUSSION TITLE
Plot lag

### DISCUSSION THREAD

**Discussion by TimTaylor** [posit-dev/positron#5174] [Q&A]
Experimenting with positron and one of the things I checked was running:
```R
plot(1:10,1:10)
```
This had a noticeable lag in comparison to running the same command in either the terminal or in RStudio (where both are pretty much instantaneous). The lag was only slight (less than 1 second) but long enough to read the "rendering plot" placeholder in the plots viewer.

I've started this as a discussion as wasn't sure if this is expected due to positron's design or possibly a bug / issue on my end?

**Comment by sharon-wang** [comment]
Hey @TimTaylor, I was able to reproduce with `plot(iris)` (a Dynamic plot). I don't think this is an issue on your end!

@seeM pinging you for more thoughts on this! Do we intentionally show "Rendering plot" before showing the plot, maybe there's a rendering bug, or could there be an underlying performance issue at play?

Static plots don't show "Rendering plot" text although it does take a moment to render the plot.

Static plot example:
```
library(ggplot2)
ggplot(midwest, aes(x=area, y=poptotal)) + geom_point()
```

So both plot types seem to take a ~1 second lag to show in the Plots pane in Positron, versus RStudio where the plot seems to show almost immediately.

**Reply by seeM** [reply]
@sharon-wang we do intentionally show "Rendering plot" before showing the plot [here](https://github.com/posit-dev/positron/blob/check-for-sqlite3-before-ipykernel-install/src/vs/workbench/contrib/positronPlots/browser/components/dynamicPlotInstance.tsx#L186).

I tested screen recording `plot(1:10,1:10)` in the terminal as well as Positron. The terminal took ~4 frames and Positron took ~30.

Is it possible that Positron is using a different DPI or other non-default plot setting? Otherwise could be a bug or poor choice on our end. I think it's worth filing as a performance issue.

**Reply by juliasilge** [reply]
I opened https://github.com/posit-dev/positron/issues/5184 and we can collect further ideas there.
